File: scripts/parse-csv.js
const fs = require('fs');
const path = require('path');

try {
  // Read CSV file
  const csvPath = path.join(__dirname, '../csv/名作ゲームデータベース - csv用.csv');
  const csv = fs.readFileSync(csvPath, 'utf-8');

  const lines = csv.trim().split('\n');
  const games = {};

  lines.slice(1).forEach((line, idx) => {
    // Simple CSV parser - handle basic cases
    const parts = [];
    let current = '';
    let inQuotes = false;

    for (let i = 0; i < line.length; i++) {
      const char = line[i];
      if (char === '"') {
        inQuotes = !inQuotes;
      } else if (char === ',' && !inQuotes) {
        parts.push(current.trim());
        current = '';
      } else {
        current += char;
      }
    }
    parts.push(current.trim());

    if (parts.length >= 5) {
      const hardware = parts[0];
      const title = parts[1];
      const year = parts[2];
      const genre = parts[4];
      const description = parts[5] || '';

      if (hardware && title && year) {
        if (!games[hardware]) games[hardware] = [];
        games[hardware].push({
          title,
          year: parseInt(year) || 0,
          genre: genre || 'UNKNOWN',
          description
        });
      }
    }
  });

  // Output as JavaScript export
  const output = `// Generated from CSV\nexport const MASTERPIECE_DB = ${JSON.stringify(games, null, 2)};\n`;

  fs.writeFileSync(path.join(__dirname, '../src/gameDatabase.js'), output, 'utf-8');
  console.log('✓ gameDatabase.js generated successfully');
} catch (err) {
  console.error('Error:', err.message);
  process.exit(1);
}

File: scripts/parse-csv.py
#!/usr/bin/env python3
import csv
import json
import os

csv_file = '/workspaces/my-history/csv/名作ゲームデータベース - csv用.csv'

games_db = {}

if os.path.exists(csv_file):
    with open(csv_file, 'r', encoding='utf-8') as f:
        reader = csv.DictReader(f)
        for row in reader:
            hardware = row.get('Hardware', '').strip()
            title = row.get('Title', '').strip()
            year = row.get('Year', '').strip()
            genre = row.get('Genre', '').strip()
            description = row.get('Description', '').strip()
            
            if hardware and title:
                if hardware not in games_db:
                    games_db[hardware] = []
                
                try:
                    year_int = int(year) if year else 0
                except:
                    year_int = 0
                
                games_db[hardware].append({
                    'title': title,
                    'year': year_int,
                    'genre': genre or 'UNKNOWN',
                    'description': description
                })

# Generate JavaScript code
output = """// Generated from CSV file
export const MASTERPIECE_DB = {
"""

for hw, games in sorted(games_db.items()):
    output += f'  {json.dumps(hw)}: [\n'
    for game in games:
        output += f"    {json.dumps(game, ensure_ascii=False)},\n"
    output += '  ],\n'

output += '};\n'

with open('/workspaces/my-history/src/gameDatabase.js', 'w', encoding='utf-8') as f:
    f.write(output)

print(f"✓ Generated gameDatabase.js with {sum(len(g) for g in games_db.values())} games across {len(games_db)} platforms")
